src/model.jl

```diff
@@ -6,3 +6,14 @@ Observe sample `x` from distribution `dist` and yield its log-likelihood value.
 function observe(dist::Distributions.Distribution, x)
     return Libtask.produce(Distributions.loglikelihood(dist, x))
 end
+
+function (instr::Libtask.Instruction{typeof(observe)})()
```

Comment:
Why is this needed? And could we dispatch on the arguments of `observe` if at some point we decide to implement add more `observe` implementations?

Reply:
This is needed because the current tape impl only tracks the direct call to `produce`, the task function calls this function, and this calls `produce`, so `produce` is not directly called by the task function and won't be recorded. This function definition makes `observe` have the same effect as `produce`.

The Instruction type is not parameterized on its input types now, I think it is possible to add this in. I will look into that later.